[hogretup] iP

hogretup opened the pull request:
MrIncredible 
“Your mind is for having ideas, not holding them.” – David Allen ([source](https://dansilvestre.com/productivity-quotes))

MrIncredible helps you keep track of everything you need to do.

It is easy to use, has a cool UI, and will help you remember everything

download it from [here](https://nus-cs2103-ay2425s1.github.io/website/schedule/week4/project.html).
double-click it.
add your tasks.
let it manage your tasks for you 😉
And it is FREE!

Features:

Managing tasks, deadlines, events
Reminders (coming soon)
If you are a Java programmer, you can use it to practice Java too. Here's the main method:

public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

hogretup committed:
Change chatbot name

hogretup committed:
Get chatbot to greet and say goodbye

hogretup committed:
Get chatbot to echo user inputs loudly

hogretup committed:
Let user add text and display back when needed

hogretup committed:
Let users mark tasks as done

hogretup committed:
Add new Task types, let chatbot handle

hogretup committed:
Add tests

hogretup committed:
Add exceptions

hogretup committed:
Add deletion of tasks

hogretup committed:
Add harddisk storage

hogretup committed:
Merge pull request #1 from hogretup/branch-Level-7

merge branch-level-7

hogretup committed:
Add date time parsing

hogretup committed:
Merge pull request #2 from hogretup/branch-Level-8

Add date time parsing

hogretup committed:
Extract out more oop

hogretup committed:
Add packages

hogretup committed:
Merge branch 'master' of https://github.com/hogretup/ip

hogretup committed:
Merge pull request #3 from hogretup/add-gradle-support

Add gradle support

hogretup committed:
add unit tests

hogretup committed:
add java docs

hogretup committed:
Merge pull request #4 from hogretup/branch-A-JavaDoc

add java docs

hogretup committed:
improve coding standard

hogretup committed:
merge master into branch

hogretup committed:
Merge pull request #5 from hogretup/branch-A-CodingStandard

improve coding standard

hogretup committed:
add finding

hogretup committed:
Merge pull request #6 from hogretup/branch-Level-9

add finding

KiKuasaurus commented:
Overall I like how well commented most of the code is, and it is very easy to follow what is going on with the code. However, some of the files do not have the same comprehensive level of documentation that makes it a bit harder to follow. Perhaps more can be done here to document the purpose of each class and method in the task and main classes.

Nihirraa commented on src/main/java/Chatbot/HarddiskStorage.java:
```diff
@@ -0,0 +1,105 @@
+import Tasks.Task;
+import Tasks.ToDo;
+
+import java.io.*;
+import java.time.format.DateTimeFormatter;
+import java.util.HashMap;
+import java.util.Map;
+import java.time.LocalDateTime;
```
The ordering of import statements need to be consistent like:
``` suggested
import java.util.HashMap;
import java.util.Map;

import java.time.format.DateTimeFormatter;
import java.time.LocalDateTime;
```

waihin26 commented on src/main/java/Chatbot/HarddiskStorage.java:
```diff
@@ -0,0 +1,105 @@
+        } catch (IOException e) {
+            System.out.println("    Error saving tasks to file.");
+        }
+    }
```
i think this code was very well written. The use of OOP throughout the whole project also made the code simple to understand and build on.

waihin26 reviewed (commented):
Overall I think the code was well done

Fluffykan commented on src/main/java/Chatbot/Parser.java:
```diff
@@ -0,0 +1,38 @@
+     * @param input the user's input as a {@code String}.
+     * @return the corresponding {@code Command} based on the input.
+     */
+    public Command parseCommand(String input) {
```
perhaps a `switch` statement could be used here in place of multiple `else if`

Fluffykan commented on src/main/java/Chatbot/TaskStorage.java:
```diff
@@ -0,0 +1,124 @@
+ * It utilizes {@code HarddiskStorage} for persistent storage and maintains a map of tasks in memory.
+ * The class supports adding, marking, retrieving, deleting, listing, and finding tasks.
+ */
+public class TaskStorage {
```
I like the clear descriptions provided in the javadoc throughout the project, it makes it easy to understand what the various methods do

Fluffykan commented on src/test/java/Chatbot/ParserTest.java:
```diff
@@ -0,0 +1,50 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class ParserTest {
```
There were ample testcases for the different possible inputs, well done! Perhaps you could try some improperly formatted inputs, eg
```java
"event project meeting /from 2024-09-01 09:00 /to 2024-09-0110:00"
"event project meeting /from  /to 2024-09-01 10:00"
"event /from 2024-09-01 09:00 /to 2024-09-01 10:00"
```

Fluffykan commented on src/main/java/Chatbot/HarddiskStorage.java:
```diff
@@ -0,0 +1,105 @@
+            String line;
+            while ((line = reader.readLine()) != null) {
+                String[] parts = line.split(" \\| ");
+                int taskId = Integer.parseInt(parts[0]);
```
perhaps there is no need to store the `taskId` of each task in the save file, instead the id could be appended to the tasks as they are loaded in from the save file since the tasks will be loaded in order of the sequence they are in in the save file

Fluffykan replied:
perhaps there is no need to store the `taskId` of each task in the save file, instead the id could be appended to the tasks as they are loaded in from the save file since the tasks will be loaded in order of the sequence they are in in the save file

hogretup committed:
add GUI

hogretup committed:
The `mrIncredible` instance is being injected into the MainWindow controller without verifying its state. Additionally, the FXML resource and controller are assumed to be correctly loaded without validation.

This can lead to runtime errors if the FXML file is missing, the controller is not initialized, or the `mrIncredible` instance is null. These issues may not be caught immediately, causing unexpected behavior.

Add `assert` statements to check that the FXML file is loaded, the `mrIncredible` instance is initialized, and the controller is properly set before injection.

Assertions provide a simple, effective mechanism to validate assumptions during development and debugging, helping catch potential issues early without affecting production performance (since they can be disabled).

Assertions should be enabled during testing and debugging using the `-ea` flag to ensure these checks are active.

hogretup committed:
Merge pull request #7 from hogretup/branch-A-Assertions

Add assertion statements

hogretup committed:
improve code quality through refactoring

hogretup committed:
Merge pull request #8 from hogretup/branch-A-CodeQuality

Improve code quality

hogretup committed:
git add contact functionality

hogretup committed:
final changes

hogretup committed:
fix command bugs and update readme
